### PR TITLE
Enable TLS when the server is created

### PR DESCRIPTION
**Pull request by aperezg**
### What has been done:

- New flag for starting server in `https://...`
- `httpServer` is now being passed as a pointer instead of a value.
Avoiding `literal copies lock value from httpServer: net/http.Server contains sync.Once contains sync.Mutex` warning
- `go-lint` variable fix

### How to test:
```
go run cmd/killgrave/main.go -imposters=internal/server/http/test/testdata/imposters -secure true
```
And make a `GET` request to https://localhost:3000/testHTTPSRequest

### To fix:
[TestBuildTLSMode](https://github.com/friendsofgo/killgrave/pull/58/files#diff-9e999a87470f8b3c2bfd32091e5a6947R129) test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,6 +127,9 @@ If you want more information about the CORS options, visit the [CORS section](#C
 
 The `watcher` configuration field is optional, with this setting you can enable hot-reloads on imposter changes. Disabled by default.
 
+The `secure` option by default is false, but if you want to run your server using TLS options with a dummy certificates, you could enable
+this option, then the mockserver be able to work with `HTTPS` protocol.
```
Comment (joanlopez) [inline README.md:131]:
```suggestion
The `secure` configuration field is optional, with this setting you can run your server using TLS options with a dummy certificate, so to make it work with the `HTTPS` protocol. Disabled by default.
```

**Review by joanlopez — APPROVED** [review]
LGTM! 🚀

**Review by marioarranzr — APPROVED** [review]
😍  thanks for taking it over! 💪

**Comment by jonthdiaz** [comment]
@aperezg do you know how can I run the version v0.4.1  use docker?

**Comment by joanlopez** [comment]
> @aperezg do you know how can I run the version v0.4.1 use docker?

The `v0.4.1` hasn't been release yet, so there's no direct way to run it. 
Anyway, if you wanna give it a try, I'd suggest to:

1. Clone the Killgrave's repository.
2. Checkout the `v0.4.1` branch.
3. Build your image locally: `docker build -t killgrave:v0.4.1 .`
4. Run Killgrave with Docker regularly but specifying the image you've built: `killgrave:v0.4.1`.

Does that work for you? Feel free to reach us if you get stuck.
I hope we release the `v0.4.1` version sooner than later.

Thanks!! 🙏🏻

**Comment by jonthdiaz** [comment]
@joanlopez Thanks so much for your answer, I run that commands but I have the next issue 
image

**Comment by joanlopez** [comment]
> @joanlopez Thanks so much for your answer, I run that commands but I have the next issue
> image

Yeah, sorry it was my fault! As the `v0.4.1` branch is still under development, it wasn't in a stable status (find additional context at https://github.com/friendsofgo/killgrave/pull/90). I also merged this PR because I guess it's part of what you're looking for the `v0.4.1` to have and because it looked approved and ready to be merged (cc/ @aperezg).

Try it again and tell us about your results. However, bear in mind that the `v0.4.1` is still a development branch, so until we release that version it could produce build failures as you experienced in the past.

Thanks!! 🙏🏻

**Pull request merged**